lib/internal/modules/cjs/loader.js

```diff
@@ -424,7 +425,19 @@ function findLongestRegisteredExtension(filename) {
   return '.js';
 }
 
-function trySelf(parentPath, request) {
+function trySelf(parent, request) {
+  if (!parent) return false;
+
+  let parentPath;
+  if (parent.filename) {
+    parentPath = parent.filename;
+  } else if (parent.id === '<repl>' || parent.id === 'internal/preload') {
+    try {
+      parentPath = process.cwd() + path.sep;
+    } catch {}
+  }
+  if (!parentPath) return false;
```

Comment:
Could we move this logic to where `trySelf` itself is called instead of inside the function itself?

It seems like this `parentPath` that is calculated can be passed in to a function call to this function fine?

Reply:
Yes, we can, would it be ok to wrap it in a function to keep the complexity in check where `trySelf` is called?

Reply:
Sure that would be fine.

Reply:
Refactored, PTAL @guybedford

Reply:
To give more context, the reason I proposed to wrap the check for the `parentPath` in a function is [this previous discussion](https://github.com/nodejs/node/pull/32261#discussion_r401201007). I don't have a strong opinion about it. I just want to follow the standard already used. I can refactor it in any way it's needed.

Reply:
Sure glad it wasn't me arguing with myself at least :P Looks good as its own function.

Reply:
:smile: